src/RuleBuilders/Architecture/Architecture.php

```diff
@@ -3,24 +3,28 @@
 
 namespace Arkitect\RuleBuilders\Architecture;
 
+use Arkitect\Expression\ForClasses\DependsOnlyOnTheseNamespaces;
 use Arkitect\Expression\ForClasses\NotDependsOnTheseNamespaces;
 use Arkitect\Expression\ForClasses\ResideInOneOfTheseNamespaces;
 use Arkitect\Rules\Rule;
 
-class Architecture implements Component, DefinedBy, Where, MayDependOnComponents, MayDependOnAnyComponent, ShouldNotDependOnAnyComponent, Rules
+class Architecture implements Component, DefinedBy, Where, MayDependOnComponents, MayDependOnAnyComponent, ShouldNotDependOnAnyComponent, ShouldOnlyDependOnComponents, Rules
 {
     /** @var string */
     private $componentName;
     /** @var array<string, string> */
     private $componentSelectors;
     /** @var array<string, string[]> */
     private $allowedDependencies;
+    /** @var array<string, string[]> */
+    private $allowedSpecificDependencies;
```

Comment:
Cosa cambia tra le `allowedDependencies` e le `allowedSpecificDependencies`?

Reply:
`allowedDependencies ` is used for creating the rule `NotDependsOnTheseNamespaces`.
`allowedSpecificDependencies` is used for creating the rule `DependsOnlyOnTheseNamespaces`.

I don't like the design now but I would like to iterate on it later. 
What do you think @pfazzi ?

Reply:
Ok, I think the implementation is a bit misleading ATM. We should iterate over it, but it could lead to a breaking change. 
Anyway in this step I would rename this variable to something more self-explaining, like `componentDependsOnlyOnTheseNamespaces`.

Reply:
I agree with you @pfazzi. 
I renamed the variable with your suggestion :)